init_code.py

```diff
@@ -1,65 +1,40 @@
 # This defines the serialization/deserialization code for keras models when used
 # in deepforge
 
+import os
+import time
+import tarfile
+import shutil
+
+import tensorflow.keras as keras
+
 import deepforge
-import keras
 
-def dump_model(model, outfile):  # Create tar ball of custom_objects and model
+
+def dump_model(model, outfile):
     # Create the tmp directory
-    import os
-    import time
     tmp_dir = outfile.name + '-tmp-' + str(time.time())
-    os.makedirs(tmp_dir)
-
-    # store any custom objects in a file
-    if hasattr(model, 'custom_objects'):
-        co_path = os.path.join(tmp_dir, 'custom_objects')
-        with open(co_path, 'wb') as f:
-            import dill as pickle
-            pickle.dump(model.custom_objects, f)
+    model.save(tmp_dir)
```

Comment:
This should actually be saving it to `outfile.name`, I believe. Is there a reason you are saving it to a tmp directory?

Reply:
The saved model format actually requires the whole directory the model and custom objects will be serialized as .proto  and its weights will be saved in the directory we specify. Hence the directory is created as mentioned in keras [documentation](https://www.tensorflow.org/guide/keras/save_and_serialize#what_the_savedmodel_contains).